package.json

```diff
@@ -467,38 +470,99 @@
         },
         "debuggers": [
             {
-                "type": "docker",
-                "label": "Docker: Node.js",
+                "type": "docker-launch",
+                "label": "Docker: Launch",
                 "configurationSnippets": [
                     {
-                        "label": "Docker: Attach to Node",
-                        "description": "Docker: Attach to Node",
+                        "label": "Docker: Launch",
+                        "description": "Docker: Launch",
```

Comment:
We'll need to think more about how these snippets are expected to work in this new model; the one here won't work at all (in particular, because there's no platform specified), and it assumes a set of generated tasks (which we can't generate at this same time).  At best, we could have a bare-minimum snippet for each supported platform (i.e. .NET Core and Node.js) and leave off the `preLaunchTask`, or set it to something obviously requiring a change by the user (e.g. `set-this-your-docker-build-task-name`).

Reply:
I think we probably should not provide snippets at all...it's highly unlikely that the user is able to hand-craft an actual, working launch config without a lot of trial and error.

Reply:
I think it's a matter of helping the user discover functionality; it the user expects to find possible debugging configurations via the "add configuration" entrypoint, then we've made it more difficult to use our features by only exposing them via the command palette.  Sure, we've perhaps mitigated that somewhat by adding the tasks/debug configuration as part of Dockerfile scaffolding, but it might still be useful to do both (even if I agree that we can't ensure an end-to-end working flow via the snippet mechanism--we should push VS Code to try to resolve that API inconsistency).

Reply:
Consider also that it's not just the debug configuration that they must trial-and-error to figure out, they have to get both the docker-build and docker-run tasks as well. I can imagine an average user failing tens of times before things work. A snippet here is just going to lead them down a path to the Dark Side (TM).